tests/apollo/test_skvbc_backup_restore.py

```diff
@@ -101,19 +106,198 @@ async def test_checkpoint_propagation_after_restarting_replicas(self, bft_networ
             verify_checkpoint_persistency=False
         )
 
+    @with_trio
+    @with_bft_network(start_replica_cmd_with_vc_timeout("20000"),
+                      selected_configs=lambda n, f, c: n == 7)
+    @with_constant_load
+    async def test_checkpoint_propagation_after_restarting_all_replicas_under_load(self, bft_network, skvbc, nursery):
+        """
+        Here we trigger a checkpoint, restart all replicas in a random order with 10s delay in-between,
+        both while stopping and starting. We verify checkpoint persisted upon restart and then trigger
+        another checkpoint. We make sure checkpoint is propagated to all the replicas.
+        1) Given a BFT network, we make sure all nodes are up
+        2) Send sufficient number of client requests to trigger checkpoint protocol
+        3) Stop all replicas in a random order (with 10s delay in between)
+        4) Start all replicas in a random order (with 10s delay in between)
+        5) Wait for view to stabilize
+        6) Send sufficient number of client requests to trigger another checkpoint
+        7) Make sure checkpoint propagates to all the replicas
+        """
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+
+        n = bft_network.config.n
+
+        self.assertEqual(len(bft_network.procs), n, "Make sure all replicas are up initially.")
+
+        current_primary = await bft_network.get_current_primary()
+
+        initial_view = 0
+
+        checkpoint_before = await bft_network.wait_for_checkpoint(replica_id=current_primary)
+
+        await skvbc.fill_and_wait_for_checkpoint(
```

Comment:
Though it is unlikely that we will get 2 checkpoints here, for consistency I think we should use _fill_and_wait_for_checkpoint_under_constant_load.